packages/env/lib/config/config.js

```diff
@@ -259,7 +259,12 @@ function withOverrides( config ) {
 				const baseUrl = new URL(
 					config.env[ envKey ].config[ configKey ]
 				);
-				baseUrl.port = config.env[ envKey ].port;
+
+				// Don't overwrite the port of WP_HOME when set.
+				if ( configKey !== 'WP_HOME' || ! baseUrl.port ) {
```

Comment:
I think we might as well apply this to all of the URLs we check here.

```suggestion
				if ( ! baseUrl.port ) {
```

Reply:
Thanks @noahtallen. I'd be happy to make this change if you think it's best. The reason that I limited this to `WP_HOME` is that I couldn't think of a legitimate reason why you would want your `WP_SITEURL` or `WP_TESTS_DOMAIN` to point to something other than the port for the WordPress environment. If you prefer limiting this to `WP_HOME` I might rewrite that conditional to be a bit more readable, like:

`if ( ! ( configKey === 'WP_HOME' && baseUrl.port ) ) {`

Reply:
> I couldn't think of a legitimate reason why you would want your WP_SITEURL or WP_TESTS_DOMAIN to point to something other than the port for the WordPress environment

Ok, that makes sense to me! If you'd like to make it more readable, that sounds good. I would also recommend a quick rebase to see if we can get the e2e tests passing!